core/src/serve_repair.rs

```diff
@@ -445,9 +464,103 @@ impl ServeRepair {
         }
     }
 
+    fn decode_requests(
+        thread_pool: &ThreadPool,
+        reqs_v: Vec<PacketBatch>,
+        epoch_staked_nodes: &Option<Arc<HashMap<Pubkey, u64>>>,
+        whitelist: &RwLockReadGuard<HashSet<Pubkey>>,
+        my_id: &Pubkey,
+        socket_addr_space: &SocketAddrSpace,
+        stats: &mut ServeRepairStats,
+        cluster_type: ClusterType,
+    ) -> Vec<RepairRequestWithMeta> {
+        thread_pool
+            .install(|| {
+                reqs_v
+                    .par_iter()
+                    .with_min_len(VERIFY_MIN_PACKETS_PER_THREAD)
+                    .flatten()
+                    .filter_map(|packet| {
+                        let request: RepairProtocol = match packet.deserialize_slice(..) {
+                            Ok(request) => request,
```

Comment:
maybe define this `|packet| {` as a separate function or lambda to avoid extensive indentation below.